File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.py
import heapq

class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        # Convert the list into a heap
        my_heap = [-n for n in nums]
        heapq.heapify(my_heap)
        
        for i in range(k):
            value = heapq.heappop(my_heap)
            if i == k - 1:
                return -value
            
        return 0

File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        Queue<Integer> maxHeap = new PriorityQueue<>();
       
        for (int num : nums) {
            maxHeap.add(num);
            
            if (maxHeap.size() > k) {
                maxHeap.poll();
            }
        }
        
        return maxHeap.peek();
    }
}